content/agent/autodiscovery/clusterchecks.md

```diff
@@ -0,0 +1,265 @@
+
+This feature is currently only supported on Kubernetes, for versions 6.9.0 and up of the Agent, and versions 1.2.0 and up of the Cluster Agent.
+
+
+## How to set it up
+
+### Cluster Agent setup
+
+This feature requires a running Cluster Agent. Refer to [the Cluster Agent setup][3] documentation and it's [Cluster Checks Autodiscovery][4] section.
+
+### Agent setup
+
+On the Datadog **Host** Agent, you need to enable the `clusterchecks` configuration provider. This can be done in two ways:
+
+- By setting the `DD_EXTRA_CONFIG_PROVIDERS` environment variable:
+
+```
+DD_EXTRA_CONFIG_PROVIDERS="clusterchecks"
+```
+
+- Or adding it to your `datadog.yaml` configuration file:
+
+```
+config_providers:
+  - name: clusterchecks
+    polling: true
+```
+
+[Restart the Agent][5] to apply the configuration change.
+
+### Custom checks
+
+Running [custom Agent Checks][6] as Cluster Checks is supported, as long as all node-based Agents are able to run it. This means your check code:
+
+- must be installed on all node-based Agents where the `clusterchecks` config provider is enabled
+- must not depend on local resources that are not accessible to all agents
+
+## Setting up Check Configurations
+
+### Static configurations in files
+
+When the IP of a given resource is constant (eg. external service endpoint, public URL...), you can pass a static configuration to the Cluster Agent as yaml files. The file name convention and syntax are the sames as static configurations on the node-based Agent, with the addition of the `cluster_check: true` line.
+
+#### Example: MySQL check on CloudSQL database
+
+After setting up your CloudSQL instance and [setting up the datadog user][7], mount a `/conf.d/mysql.yaml` file in the Cluster Agent container with the following contents:
+
+```yaml
+cluster_check: true
+init_config:
+instances:
+  - server: '<PRIVATE_IP_ADDRESS>'
+    port: 3306
+    user: datadog
+    pass: '<YOUR_CHOSEN_PASSWORD>'
+```
+
+The `cluster_check` field will inform the Cluster Agent to delegate this check to one node-based Agent.
+
+### Template Source: Kubernetes Service Annotations
+
+As you can [annotate Kubernetes Pods][8], you can annotate Services with the following syntax:
+
+```yaml
+  ad.datadoghq.com/service.check_names: '[<CHECK_NAME>]'
+  ad.datadoghq.com/service.init_configs: '[<INIT_CONFIG>]'
+  ad.datadoghq.com/service.instances: '[<INSTANCE_CONFIG>]'
+```
+
+The `%%host%%` [template variable][9] is supported and will be replaced by the service's IP. The `kube_namespace` and `kube_service` tags will be added to the instance.
+
+#### Example: HTTP check on an nginx-backed service
+
+The following Service definition will expose the Pods from the `my-nginx` deployment and run an [HTTP check][10] to measure the latency of the load-balanced service:
+
+```
```

Comment:
dunno if we color stuff in our docs, but in case we do (or plan on doing it in the future), we could add `yaml` after the 3 backquotes here.